From: Nico Berlee <nico.berlee@on2it.net>

Incorporates patches inspired on:
 - https://patchwork.kernel.org/project/linux-rockchip/patch/20240401081302.942742-1-liujianfeng1994@gmail.com/
 - https://gitlab.collabora.com/hardware-enablement/rockchip-3588/linux/-/commit/3f2d6dea70fa69a7b0531e64f3668f588d2da2c3


Property startup-delay-us is copied from vendor dts and it will make the
kernel not detect stmmac not the first time sometimes. Making startup
slower than actually needed. Talos does a rescan after modules are loaded,
and stmmac is loaded then.


Mark the primary PMIC as system-power-controller, so that the
system properly shuts down on poweroff.

Signed-off-by: Nico Berlee <nico.berlee@on2it.net>

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-turing-rk1.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-turing-rk1.dtsi
index 7bf97fe..01b30b5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-turing-rk1.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-turing-rk1.dtsi
@@ -44,7 +44,6 @@ vcc3v3_pcie30: vcc3v3-pcie30-regulator {
 		gpios = <&gpio2 RK_PC5 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&vcc3v3_pcie30_en>;
-		startup-delay-us = <5000>;
 	};

 	vcc5v0_sys: vcc5v0-sys-regulator {
@@ -301,6 +300,7 @@ pmic@0 {
 		compatible = "rockchip,rk806";
 		spi-max-frequency = <1000000>;
 		reg = <0x0>;
+		system-power-controller;

 		interrupt-parent = <&gpio0>;
 		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
